src/core/contract.service.ts

```diff
@@ -31,6 +32,7 @@ export class ContractService {
 
   constructor(
     @InjectRepository(Controller) private controllerRepo: Repository<Controller>,
```

Comment:
You can remove it. All controller operations should be limited in ControllerService